elastic/kibana
# [Stack Monitoring] Add support for beats datastream patterns

## Summary
Closes https://github.com/elastic/kibana/issues/146686

Update apm and beats queries to read from the Beat package data streams created in https://github.com/elastic/integrations/pull/4708. Also updates health API to fetch from the data streams.

API tests follow up in https://github.com/elastic/kibana/pull/147755

### Testing
That's a fairly heavy workflow. I'm investigating ways to make that easier

- build `beat` package with `elastic-package build`. The package is not published yet so you'll need to pull https://github.com/elastic/integrations/pull/4708 locally
- start a stack `elastic-package stack up -v -d --version 8.7.0-SNAPSHOT`. Make sure you do this within `integrations` repo so it picks up the previously built package
- start a beat service with `elastic-package service up -v --variant metricbeat_8.7.0`
- install both elasticsearch and beat packages from the kibana started by the stack command (`https://localhost:5601`). Nothing shows up in SM if we don't have elasticsearch data.
  - both packages are prerelease versions and we need to explicitly tell Integrations plugin to show them up[1]
  - elasticsearch hosts is `https://elasticsearch:9200` and beat _should be_ `http://elastic-package-service_beat_1:5066` but it may differ depending on your docker version
- start a Kibana with this branch connected to the elasticsearch instance from the `stack up` command. [see howto](https://github.com/elastic/observability-dev/blob/main/docs/infra-obs-ui/stack-monitoring_integration-packages.md#connecting-a-local-kibana)
- navigate to Stack Monitoring and verify the metricbeat is properly monitored
- start an apm server with `elastic-package service up -v --variant apm_8.7.0`
- navigate to Stack Monitoring and verify apm-server is properly monitored


[1]
<img width="853" alt="Screenshot 2022-12-01 at 00 20 20" src="https://user-images.githubusercontent.com/5239883/204929352-c7656679-f88c-4013-b1f8-cf7c67d4c830.png">
